### PR TITLE
Add Content for Jenkins DAG Only Deploy

### LINKED ISSUE #1749: Jenkins `--dags` instructions
**Affected Document:** https://docs.astronomer.io/astro/ci-cd#jenkins

**Problem Type (Priority):** Important

- [ ] Error (Critical): This document has existing information that's incorrect and prevents readers from completing the documented setup (Add the `error/critical` label to this issue)
- [ ] Misleading (Important): This document could benefit from additional context, guidance, or clarity (Add the `error/FYI` label to this issue)
- [ ] Typo (Low): Any issue related to spelling/ punctuation/ formatting (Add the `error/typo` label to this issue)

**Suggested Change (include links to relevant existing docs, support tickets, messages, or external content):** Would like to see instructions for implementing `astro deploy --dags` with Jenkins syntax similar to what we provide for GitHub Actions in this section: https://docs.astronomer.io/astro/ci-cd#github-actions-dag-based-deploy

**Additional Notes:** Requested by Idelic

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1749

### CODE REVIEW

File: astro/ci-cd.md [hunk 1]
```diff
@@ -757,6 +757,53 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+              '''
+          }
+        }
+        }
+         post {
+           always {
+           cleanWs()
+      }
+     }
+     }</code></pre>
```
Comment (jwitz) [inline astro/ci-cd.md:804]:
```suggestion
     }`</code></pre>
```

File: astro/ci-cd.md [hunk 2]
```diff
@@ -757,6 +757,53 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+
+    - `ASTRONOMER_KEY_ID`: Your Deployment API key ID
+    - `ASTRONOMER_KEY_SECRET`: Your Deployment API key secret
+    - `ASTRONOMER_DEPLOYMENT_ID`: The Deployment ID of your production deployment
+
+    Be sure to set the values for your API credentials as secret.
+
+2. At the root of your Git repository, add a [Jenkinsfile](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/) that includes the following script:
+
+    <pre><code parentName="pre">pipeline {
```
Comment (jwitz) [inline astro/ci-cd.md:774]:
```suggestion
    <pre><code parentName="pre">` pipeline {
```

File: astro/ci-cd.md [hunk 3]
```diff
@@ -757,6 +757,53 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+           }
+          }
+          steps {
+              checkout scm
+              sh '''
+                curl -LJO https://github.com/astronomer/astro-cli/releases/download/v${siteVariables.cliVersion}/astro_${siteVariables.cliVersion}_linux_amd64.tar.gz
+                tar -zxvf astro_${siteVariables.cliVersion}_linux_amd64.tar.gz astro && rm astro_${siteVariables.cliVersion}_linux_amd64.tar.gz
+                files=($(git diff-tree HEAD --name-only --no-commit-id))
+                find="dags"
+                if [[ ${files[*]} =~ (^|[[:space:]])"$find"($|[[:space:]]) && ${#files[@]} -eq 1 ]]; then
```
Comment (jwitz) [inline astro/ci-cd.md:790]:
```suggestion
                if [[ ${siteVariables.jenkinsenv1} =~ (^|[[:space:]])"$find"($|[[:space:]]) && ${siteVariables.jenkinsenv2} -eq 1 ]]; then
```
@kwcanuck To fix the build issue here, add the following lines after `jenkinsev to `src/versions.js`: 

```
jenkinsenv1: '${files[*]}',
jenkinsenv2: '${#files[@]}',
```

**Review by pohek321 — APPROVED** [review]
Added a commit to remove the `-f` from the other jenkins scripts. @jwitz if you are okay with that commit then feel free to merge.

**Comment by kwcanuck** [comment]
@jwitz with your approval, I can merge the update.

**Review by jwitz — APPROVED** [review]

**Pull request merged**